arkouda-env-dev.yml

```diff
@@ -12,10 +12,10 @@ dependencies:
   - versioneer
   - matplotlib>=3.3.2
   - h5py>=3.7.0
-  - hdf5==1.12.2
+  - hdf5>=1.12.2
```

Comment:
I'm not sure if we should unpin hdf5 or pin to the new makefile version. My guess is this was done to ensure the `make deps` and conda env versions don't get out of step